compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -90,6 +104,16 @@ struct ConvertToROCDLPass final
   void runOnOperation() override {
     ModuleOp m = getOperation();
 
+    m.walk(
+        [&](Operation *op) {
```

Comment:
Does this capture anything? If we only need the capture for `this`, it seems like we could make `validateDataTypes` a free/static function instead